pyproj/_crs.pyx

```diff
@@ -391,15 +500,56 @@ cdef class _CRS:
         """
         Returns
         -------
-        pyproj.CRS: The datum as a CRS.
+        Datum: The datum.
         """
-        cdef PJ *projobj = proj_crs_get_datum(self.projctx, self.projobj)
+        if self._datum is not None:
+            return self._datum
+        self._datum = Datum.create(self.projctx, self.projobj)
+        return self._datum
+
+    @property
+    def source_crs(self):
+        """
+        Returns
+        -------
+        CRS: The source CRS.
+        """
+        if self._source_crs is not None:
+            return self._source_crs
+        cdef PJ * projobj
+        projobj = proj_get_source_crs(self.projctx, self.projobj)
         if projobj == NULL:
             return None
```

Comment:
Also assign the None case to `self._source_crs` so we don't have to retry every time you access this property? (since CRS is immutable, this should be fine?)

Reply:
Good catch